src/main/java/com/example/bookstore/service/CategoryService.java

```diff
@@ -0,0 +1,17 @@
+package com.example.bookstore.service;
+
+import com.example.bookstore.dto.category.CategoryDto;
+import com.example.bookstore.dto.category.CreateCategoryRequestDto;
+import java.util.List;
+
+public interface CategoryService {
+    List<CategoryDto> findAll();
+
+    CategoryDto getById(Long id);
+
+    CategoryDto save(CreateCategoryRequestDto dto);
+
+    CategoryDto update(Long id, CreateCategoryRequestDto dto);
+
+    void deleteById(Long id);
```

Comment:
it never hurts to return some value from a method even if we do not need one, it can be useful to ensure that everything actually works as expected, and tests-wise, but that is up to you

Reply:
i return http response status no content in category controller on DELETE endpoint @ResponseStatus(HttpStatus.NO_CONTENT)

Reply:
that approach is also Ok, I was talking about returning some value in that method, but as I said it is up to you :)